File: server/src/main/java/com/example/demo/models/RecipeComment.java
package com.example.demo.models;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Entity
@Table(name = "recipe_comment")
@Data
@AllArgsConstructor
@NoArgsConstructor
public class RecipeComment {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "message", nullable = false)
    private String message;

    @ManyToOne
    @JsonIgnore
    @JoinColumn(name = "recipe_id", nullable = false)
    private Recipe recipe;

    @ManyToOne
    @JoinColumn(name = "author_id")
    @JsonIgnoreProperties("recipes")
    private User author;

    // Read only
    @Column(name = "created_at", columnDefinition = "TIMESTAMP(3) DEFAULT CURRENT_TIMESTAMP")
    private LocalDateTime createdAt;

    @Column(name = "updated_at", columnDefinition = "TIMESTAMP(3) DEFAULT CURRENT_TIMESTAMP")
    private LocalDateTime updatedAt;
}

File: server/src/main/kotlin/com/example/demo/recipe/domain/recipecomment/RecipeComment.kt
package com.example.demo.recipe.domain.recipecomment

import com.example.demo.user.domain.Recipe
import com.example.demo.user.domain.User
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import jakarta.persistence.Table
import java.util.Date
import org.springframework.data.annotation.CreatedDate

@Entity
@Table(name = "recipe_comments")
data class RecipeComment(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY) var id: Int? = null,
    @Column(name = "message", nullable = false, length = 500) var message: String,
    @ManyToOne @JoinColumn(name = "recipe_id", nullable = false) val recipe: Recipe,
    @ManyToOne @JoinColumn(name = "user_id", nullable = false) val author: User,
    @CreatedDate @Column(name = "created_at", nullable = false) var createdAt: Date = Date(),
)
